src/language-js/needs-parens.js

```diff
@@ -945,6 +945,31 @@ function needsParens(path, options) {
           }
         }
       }
+
+      if (
+        parent.type === "TaggedTemplateExpression" &&
+        (node.type === "OptionalMemberExpression" ||
+          node.type === "OptionalCallExpression")
+      ) {
+        return true;
+      }
+
+      if (
```

Comment:
I don't think this block is needed, simple `node.type === "ChainExpressin"` should be enough.

Reply:
line 952 for babel parser. And line 957 for another parsers.

#### babel parser structure
![스크린샷 2024-07-23 오전 1 47 43](https://github.com/user-attachments/assets/a59a836d-c2ed-4b63-9332-0b37a9e20802)

#### acorn parser structure
![스크린샷 2024-07-23 오전 1 47 30](https://github.com/user-attachments/assets/ef0e466c-233a-45d8-a9b8-52c07299401f)

Reply:
I'm aware the difference, we don't need check children.

Reply:
Babel parser can be noticed by OptionalMemberExpression or OptionalCallExpression at the top if there is an optional chaining.
But for the rest of the parser, i have to explore the optional properties of the child channings. Am I misinterpreting this?

Reply:
ChainExpressin always contains optional call or member expression, so we don't need check the descent.

Reply:
<img width="1122" alt="스크린샷 2024-07-23 오후 4 31 53" src="https://github.com/user-attachments/assets/bd197dfc-4492-4445-9e0d-0fd623e4ce55">
Logs show that `node.type` is never optional call or member expression.

Reply:
Apply this https://github.com/prettier/prettier/pull/16500/files#r1687578916, you'll see what I mean.